reframe/utility/__init__.py

```diff
@@ -304,6 +304,27 @@ def attrs(obj):
     return ret
 
 
+def is_trivially_callable(fn, *, non_def_args=0):
```

Comment:
Yes, I figured this and it's useful, but I suggest to adapt a bit the documentation of the function because now it reads as "A trivially callable object is one that can be called without providing any additional arguments to its call method," which is a bit contradictory 😭